app/collectors/instance.scala

```diff
@@ -26,43 +23,47 @@ class InstanceCollectorSet(accounts: Accounts, prism: Prism) extends CollectorSe
 
 case class AWSInstanceCollector(origin:AmazonOrigin, resource: ResourceType, crawlRate: CrawlRate, prism: Prism) extends Collector[Instance] with Logging {
 
-  val client: AmazonEC2 = AmazonEC2ClientBuilder.standard()
-    .withCredentials(origin.credentials.provider)
-    .withRegion(origin.awsRegion)
-    .withClientConfiguration(AWS.clientConfig)
+  val client: Ec2Client = Ec2Client
+    .builder()
+    .credentialsProvider(origin.credentials.providerV2)
+    .region(origin.awsRegionV2)
+    .overrideConfiguration(AWS.clientConfigV2)
     .build()
 
-
-  def getInstances:Iterable[(AWSReservation, AWSInstance)] = {
-    PaginatedAWSRequest.run(client.describeInstances)(new DescribeInstancesRequest())
+  def getInstances:Iterable[(AwsReservation, AwsInstance)] = {
+    val reservations = client.describeInstancesPaginator(DescribeInstancesRequest.builder.build).reservations.asScala
+    for {
+      reservation <- reservations
+      instance <- reservation.instances.asScala
+    } yield (reservation, instance)
   }
 
   def crawl:Iterable[Instance] = {
     getInstances.map { case (reservation, instance) =>
       Instance.fromApiData(
-        arn = s"arn:aws:ec2:${origin.region}:${origin.accountNumber.getOrElse(reservation.getOwnerId)}:instance/${instance.getInstanceId}",
-        vendorState = Some(instance.getState.getName),
-        group = instance.getPlacement.getAvailabilityZone,
+        arn = s"arn:aws:ec2:${origin.region}:${origin.accountNumber.getOrElse(reservation.ownerId)}:instance/${instance.instanceId}",
+        vendorState = Some(instance.state.nameAsString),
+        group = instance.placement.availabilityZone,
         addresses = AddressList(
-          "public" -> Address(instance.getPublicDnsName, instance.getPublicIpAddress),
-          "private" -> Address(instance.getPrivateDnsName, instance.getPrivateIpAddress)
+          "public" -> Address(instance.publicDnsName, instance.publicIpAddress),
+          "private" -> Address(instance.privateDnsName, instance.privateIpAddress)
         ),
-        createdAt = new DateTime(instance.getLaunchTime),
-        instanceName = instance.getInstanceId,
+        createdAt = instance.launchTime,
```

Comment:
oooh, naughty - it never used a `Try`. Absolute proof that it wasn't needed though!!! 😂